server/cluster/cluster.go

```diff
@@ -444,6 +446,39 @@ func (c *RaftCluster) RemoveSuspectRegion(id uint64) {
 	c.suspectRegions.Remove(id)
 }
 
+// AddSuspectKeyRanges add the key ranges with the its ruleID as the key
+// The instance of each keyRange is like following format:
+// [2][]byte: start key/ned key
+func (c *RaftCluster) AddSuspectKeyRanges(key string, keyRanges [][2][]byte) {
+	c.Lock()
+	defer c.Unlock()
+	c.suspectKeyRanges.Put(key, keyRanges)
```

Comment:
I wonder if it is necessary to put 2 ranges at the same time. Can we store `[2][]byte` instead?

Reply:
I use `[][2][]byte` due to they have the same key, generated by rule' StoreKey and the value here is newVersionStartKey/newVersionEndKey/oldVersionStartKey/oldVersionEndKey.

Reply:
updated.